File: src/frontend/src/app/_dtos/rule.ts
export interface IRuleBase {
	name: string;
	limit: number;
}

export class RuleBase implements IRuleBase {
	name: string = "";
	limit: number = 150;

	constructor(rule?: IRuleBase) {
		if (rule) {
			this.name = rule.name;
			this.limit = rule.limit;
		}
	}
}

export interface IRule extends IRuleBase{
	id: number;
}

export class Rule extends RuleBase implements IRule {
	id: number;

	constructor(rule: IRule) {
		super(rule);

		this.id = rule.id;
	}
}

File: src/backend/schemas/rule.py
from typing import Optional

from pydantic import BaseModel


# Shared properties
class RuleBase(BaseModel):
    name: str
    limit: int


class RuleCreate(RuleBase):
    pass


class RuleInDBBase(RuleBase):
    id: Optional[int] = None

    class Config:
        orm_mode = True


# Additional properties to return via API
class Rule(RuleInDBBase):
    pass


class RuleData(BaseModel):
    rule_id: int
